usr/share/rear/skel/default/etc/scripts/unlock-opal-disks

```diff
@@ -29,28 +30,53 @@ function enter_plymouth_shutdown_mode() {
 
 function display_message() {
     local message="${1:?}"
-
     (use_plymouth && plymouth display-message --text="$message") || echo -e "\n$message"
 }
 
+function hide_message() {
+    #local message="${1:?}"
+    #use_plymouth && plymouth hide-message --text="$message"
+    # hide-message doesn't work, at least with two-step plugin
+    use_plymouth && plymouth display-message --text=""
+    return 0
+}
+
 function ask_for_password() {
     local prompt="${1:?}"  # a colon will be appended implicitly
+    # $2 - will accept an empty password if this arg passed
     # asks for a password, setting the variable $password.
+    local msg_req="Please enter a non-empty password"
 
     while true; do
         if ! password="$(use_plymouth && plymouth ask-for-password --prompt="$prompt")"; then
             echo ""
             read -r -s -p "$prompt: " password 2>&1
```

Comment:
In general we use the UserInput function instead of plain 'rear'
in particular because the UserInput function allows to run ReaR
unattended which is required for automated environments, cf.
https://github.com/rear/rear/wiki/Coding-Style#it-should-be-possible-to-run-rear-unattended

See the UserInput function description in
rear/lib/_input-output-functions.sh
how to get confidential user input
https://github.com/rear/rear/blob/rear-2.7/usr/share/rear/lib/_input-output-functions.sh#L982

Reply:
Ah - yes - 'read' is right here.
I overlooked that is is a recovery system setup script.
But plain 'read' without timeout is problematic, see
https://github.com/rear/rear/issues/2955#issuecomment-1465642166